src/shims/unix/linux/eventfd.rs

```diff
@@ -30,13 +32,52 @@ struct Event {
     counter: u64,
     is_nonblock: bool,
     clock: VClock,
+    epoll_events: Vec<Weak<EpollEvent>>,
 }
 
 impl FileDescription for Event {
     fn name(&self) -> &'static str {
         "event"
     }
 
+    fn check_readiness<'tcx>(&self, ecx: &mut MiriInterpCx<'tcx>) -> InterpResult<'tcx, Vec<u32>> {
+        // We only check the status of epollin and epollout flag for eventfd. If other event flags
+        //         // need to be supported in future, the check should be added here.
+        let epollin = ecx.eval_libc_u32("EPOLLIN");
+        let epollout = ecx.eval_libc_u32("EPOLLOUT");
+        let mut ready_flags = Vec::new();
+        // Check if it is readable.
+        if self.counter != 0 {
+            ready_flags.push(epollin);
+        }
+        // Check if it is writable.
+        if self.counter != MAX_COUNTER {
+            ready_flags.push(epollout);
+        }
+        Ok(ready_flags)
+    }
+    fn update_readiness<'tcx>(&self, ready_flags: Vec<u32>) -> InterpResult<'tcx> {
+        // TODO: socketpair and eventfd has the same update_readiness, possible to deduplicate?
+        for event in &self.epoll_events {
```

Comment:
Instead of having a `update_readiness` function, you could add a function that returns the epoll_events list. Then the current update logic can live outside of the file descriptors